### PR TITLE
Add postgres repository

### PR DESCRIPTION
**Pull request by whisper-bye**
add postgres repository, for example

```
create table apis
(
	json jsonb
)
;

create table oauth
(
	json jsonb
)
;
```

```
[database]
  dsn = "postgres://postgres@localhost:5432/janus"
```

### CODE REVIEW

File: Gopkg.lock [hunk 1]
```diff
@@ -213,6 +213,15 @@
   revision = "f611eb38b3875cc3bd991ca91c51d06446afa14c"
   version = "v1.3.0"
 
+[[projects]]
+  branch = "master"
+  name = "github.com/lib/pq"
```
Comment (vgarvardt) [inline Gopkg.lock:218]: I'd rather used https://github.com/jackc/pgx instead of `lib/pq`

File: pkg/api/postgres_repository.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	log "github.com/sirupsen/logrus"
+	"time"
```
Comment (vgarvardt) [inline pkg/api/postgres_repository.go]: format inports pls

**Review by vgarvardt — COMMENTED** [review]
For relational DB it makes sense to use DB features, e.g. extract proxy/oauth name to a separate column with primary key to let the DB validate name uniqueness and make update/delete queries simplier.

File: Gopkg.lock (hunk 1 above)
Comment (rafaeljesus) [reply Gopkg.lock:218]: @whisper-bye thanks for your contribution, `upper/db` is too much for our use case, we need a tinier one as. In addition to that ideally if postgres being supported we don't need the polling implementation given PG has notify/listen mechanism, so the PG implementation would have this feature

Comment (whisper-bye) [reply Gopkg.lock:218]: thanks, close this PR

**Pull request closed without merging**